pkg/controllers/state/cluster.go

```diff
@@ -53,10 +55,11 @@ type Cluster struct {
 
 func NewCluster(client client.Client, cp cloudprovider.CloudProvider) *Cluster {
 	c := &Cluster{
-		kubeClient:    client,
-		cloudProvider: cp,
-		nodes:         map[string]*Node{},
-		bindings:      map[types.NamespacedName]string{},
+		kubeClient:     client,
+		cloudProvider:  cp,
+		nominatedNodes: cache.New(10*time.Second, 10*time.Second),
```

Comment:
Do you mind adding a comment about why 10 seconds here? IIUC, this is the amount of time we'd expect the kube scheduler to kick in after a node is ready.